### ISSUE TITLE
Ability to clone/checkout volume tree

### ISSUE THREAD

**Issue by jamesmoey** [ClusterHQ/dvol#8]
It would be very useful to be able to clone/checkout volume tree from a remote machine. Similiar to git clone functionality.

**Comment by lukemarsden** [comment]
Thanks for the feedback! What use case do you have in mind for this? I'd love to know in a bit more detail how you would use this feature.

**Comment by jamesmoey** [comment]
Given this ability. The committed volume can be share between developers. No sure how merge and conflict is going to be manage. Basically we need a way to export and import the volume tree or/and a volume commit, so it can be shared between developer.

**Comment by zitrusmedia** [comment]
yeah! that would be a really cool feature. push and clone / checkout data volumes to share them easily between teammates.

**Comment by Crashthatch** [comment]
Correct me if I'm wrong, but wouldn't this functionality make migrations trivial? Run dvol commit / dvol push on the old live server, then do dvol pull on the new server (at the same time you do docker-up for the source & processes) and you'd have an identical copy up & running.

Would also be good for going live (creating fixtures that are not source-controlled etc.), for creating off-site backups, and for creating a snapshot of live database on a development machine. Obviously these things can be done manually, but doing it at the volume level would be nice in that you don't need to find a process for exporting from all the different components. Databases, uploaded media, log files, they'd all be transferred in the same way. (analogous to how docker abstracts away the details of what processes etc. are contained inside the image).

I know flocker already does this within a cluster, but for development this could be really handy.

**Comment by lukemarsden** [comment]
Hey @jamesmoey, @loeffert, @Crashthatch - thanks so much for the feedback! I agree all of these use cases are really cool. We plan to start working on this feature in the next few weeks, I'll update this issue when we have any progress / a design proposal / a beta release to start playing with. Thanks!

**Comment by ph-One** [comment]
Any updates on this enhancement, fellas? I was just about to put in a request for this, but I see others already beat me to it. I actually expected this to be one of the core features of `dvol` -- cloning. It would be great to be able to [easily] share volumes with others in a clone, push, or pull fashion.
:beers:
